lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -291,12 +285,12 @@ public function setParameter($key, $value, $type = null)
      *         ));
      * </code>
      *
-     * @param mixed[]        $params The query parameters to set.
-     * @param int[]|string[] $types  The query parameters types to set.
+     * @param array<string|int, mixed> $params The query parameters to set.
+     * @param array<string|int, mixed> $types  The query parameters types to set.
```

Comment:
The types of `$params` and `$types` keys cannot be mixed (should be `array<…>|array<…>`).